File: APCS Final Project/src/befunge/Stack.java
package befunge;

public class Stack {
	
	private StackElement top;

	public Stack() {
		top = null;
	}
	
	public void push(int i) {
		StackElement element = new StackElement(i);
		if (top == null) {
			top = element;
		}
		else {
			element.setNext(top);
			top = element;
		}
	}
	
	public int peek() {
		return top.getData();
	}
	
	public int pop() {
		StackElement ret = top;
		top = top.getNext();
		return ret.getData();
	}
	
	public StackElement getTop() {
		return top;
	}

	public void setTop(StackElement top) {
		this.top = top;
	}
	
	public boolean isEmpty() {
		return top == null;
	}
	
	public static class StackElement {
		
		private int data;
		
		private StackElement next;
		
		public StackElement() {
			this(0, null);
		}

		public StackElement(int data) {
			this(data, null);
		}
		
		public StackElement(int data, StackElement next) {
			this.setData(data);
			this.setNext(next);
		}

		public int getData() {
			return data;
		}

		public void setData(int data) {
			this.data = data;
		}

		public StackElement getNext() {
			return next;
		}

		public void setNext(StackElement next) {
			this.next = next;
		}
		
		public String toString() {
			return this.getData() + "";
		}

	}
	
	public String toString() {
		if (top == null) return "";
		StringBuilder str = new StringBuilder();
		StackElement current = top;
		while (current != null) {
			str.insert(0, current.getData() + " ");
			current = current.getNext();
		}
		return str.toString();
	}

}

File: Befunge/Befunge/Stack.cs
using System;
using System.Text;

namespace Befunge
{
	public class Stack
	{

		private StackElement top;

		public Stack()
		{
			top = null;
		}

		public void Push(int i)
		{
			StackElement element = new StackElement(i);
			if (top == null)
			{
				top = element;
			}
			else
			{
				element.Next = top;
				top = element;
			}
		}

		public int Peek()
		{
			if (IsEmpty())
				return 0;
			return top.Data;
		}

		public int Pop()
		{
			StackElement ret = top;
			top = top.Next;
			return ret.Data;
		}

		public StackElement GetTop()
		{
			return top;
		}

		public void SetTop(StackElement top)
		{
			this.top = top;
		}

		public bool IsEmpty()
		{
			return top == null;
		}

		public class StackElement
		{

			private int data;
			public int Data
			{
				get
				{
					return data;
				}
			}

			private StackElement next;
			public StackElement Next
			{
				get
				{
					return next;
				}
				set
				{
					next = value;
				}
			}

			public StackElement(int data, StackElement next)
			{
				this.data = data;
				this.next = next;
			}

			public StackElement(int data)
				: this(data, null)
			{

			}

			public StackElement()
				: this(0, null)
			{

			}

			public override string ToString()
			{
				return "" + data;
			}

		}

		public override string ToString()
		{
			if (top == null) return "Stack: ";
			StringBuilder str = new StringBuilder("Stack: ");
			StackElement current = top;
			while (current != null)
			{
				str.Insert(7, current.Data + " ");
				current = current.Next;
			}
			return str.ToString();
		}

	}
}